zeebe/exporters/elasticsearch-exporter/src/main/java/io/camunda/zeebe/exporter/RecordIndexRouter.java

```diff
@@ -40,7 +41,18 @@ final class RecordIndexRouter {
    */
   String indexFor(final Record<?> record) {
     final Instant timestamp = Instant.ofEpochMilli(record.getTimestamp());
-    return (indexPrefixForValueType(record.getValueType()) + INDEX_DELIMITER)
+
+    final var recordVersion = SemanticVersion.parse(record.getBrokerVersion());
+    var indexVersion = SemanticVersion.parse(VersionUtil.getVersionLowerCase()).orElseThrow();
+
+    // if the record version is less than the current version use that as the record should be
+    // written to old versioned index.
+    if (recordVersion.isPresent() && recordVersion.get().compareTo(indexVersion) < 0) {
+      indexVersion = recordVersion.get();
+    }
+
+    return (indexPrefixForValueType(record.getValueType(), indexVersion.toString().toLowerCase())
+            + INDEX_DELIMITER)
         + formatter.format(timestamp);
```

Comment:
❌ I think we should "simply" always use the Record version, no?

Reply:
I agree. The version is set on the record. If it's older we should write it to the older index. If it's not older, it should be the same as the version that comes out of `VersionUtil` anyway. So the version comparison is redundant.

Reply:
True the record version is what we should write to but there is a lot of tests which will fail as a result of not checking if the recordVersion is a valid semantic version as the factory which generates records will fill it with a random value. `

I agree though I'll work through the tests

<img width="1145" alt="image" src="https://github.com/user-attachments/assets/fd178e23-363c-4944-8ae2-b97a0928ea34">